mail_body_notified_partner/tests/test_mail_notification.py

```diff
@@ -0,0 +1,67 @@
+# © 2016 ACSONE SA/NV <https://acsone.eu>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import odoo.tests.common as common
+
+
+class TestMailNotification(common.TransactionCase):
+    def setUp(self):
+        super().setUp()
+        self.sender = self.env["res.partner"].create(
+            {
+                "name": "Test sender",
+                "email": "sender@example.com",
+            }
+        )
+        self.recipient = self.env["res.partner"].create(
+            {
+                "name": "Test recipient",
+                "email": "recipient@example.com",
+            }
+        )
+        self.message = self.env["mail.message"].create(
+            {
+                "subject": "Message test",
+                "author_id": self.sender.id,
+                "email_from": self.sender.email,
+                "message_type": "comment",
+                "model": "res.partner",
+                "res_id": self.recipient.id,
+                "partner_ids": [(4, self.recipient.id)],
+                "body": "<p>This is a test message</p>",
+            }
+        )
+
+        self.record = self.recipient
+
+        self.rdata = [
+            {
+                "type": "customer",
+                "id": self.recipient.id,
+                "groups": [None],
+                "active": True,
+                "share": True,
+                "notif": "email",
+            }
+        ]
+        self.force_send = True
+        self.send_after_commit = True
+        self.model_description = False
+        self.mail_auto_delete = True
+
+    def test_get_signature_footer(self):
+        rep = self.recipient._notify(
```

Comment:
This method doesn't seem to exist anymore in v14. You will probably need to adapt a bit this test code.

Reply:
Thank you for your comment. Unfortunately this is out of my knowledge... could you please suggest the fix ?

Reply:
I don't know the fix. I'd have to dig deep in the code of Odoo 16 to know that. However, how would I do it if I were you?

First thing would be to understand how unit tests work. It seems to me you never did these kind of tests, which is fine, but if you're migrating modules, this is an essential part of the migration work. Keep in mind that, if you've used modules and they work fine, it's because many others have devoted their time to write these tests. If you learn to do the same, you'll get to the next level! So, try starting by [reading docs about tests](https://www.odoo.com/documentation/14.0/developer/howtos/rdtraining/E_unittest.html). Install the module locally (I imagine you already did, but just in case), and run the tests. See the failure and fix it.

Now... how to fix it? As you can see, [in v12 `_notify()` was declared here](https://github.com/odoo/odoo/blob/c53081f10befd4f1c98e46a450ed3bc71a6246ed/addons/mail/models/res_partner.py#L89). However, [in v13 that method no longer exists](https://github.com/odoo/odoo/blob/55fbbfc26d0568b1b1c53ccf7700b703c8b1bb8b/addons/mail/models/res_partner.py). So I'd go and review the commits (or PRs) between v12 and v13 and see which one removed that method. It should lead you to the explanation about why it was removed and what replaces that functionality.

Once you find the replacement, just adapt this call and you'll get it fixed! 😃